### PR TITLE
fix: splitting jsx text correctly

### PR DESCRIPTION
**Pull request by yuliaHope**
Fixes: #4941
- [x] I’ve added tests to confirm my change works.
- [x] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: tests/jsx-text-wrap/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -474,23 +513,9 @@ x = (
...
-    after
-    {stuff}
-    after
-    {stuff}
-    {stuff}
-    {stuff}
-    after
-    {stuff}
-    after
+    {stuff}after{stuff}after
```
Comment (lydell) [inline tests/jsx-text-wrap/__snapshots__/jsfmt.spec.js.snap]: Why is there one line with only a single `{stuff}` in it? This is a weird edge case so it doesn’t matter much, but I’d like to understand how things work.

Comment (yuliaHope) [reply tests/jsx-text-wrap/__snapshots__/jsfmt.spec.js.snap]: Because we have three jsx expressins one after another. The same [here](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAPABAXnQHgCYCWAbgHzByoAOATnAM50HQCMAvuVbQ01AEzsUa9RhADMrHAHpCpEABoQESjB51koAIbVqEAO4AFLQjUoNRCATzyQAI2oawAazgwAypQcEoAc2QxqAK5wCl50cNQw+vbeALYayABmGgA2YQoAVnSoAEL2Ti6uGjFwADJecIkpaSAe1GHUyLYaNgCeydDWNF4wAOqWMAAWyAAcAAwKNBBhPfaUjUL1RBUKtACOAQS0URqx8UhJqcEgYTEEfoFHjD7JcACKARDwlYcKMM19eIPIvK-2BMlebwAYQgMTijSg0GWIACYQAKs0TAcwqxWEA)

Comment (yuliaHope) [reply tests/jsx-text-wrap/__snapshots__/jsfmt.spec.js.snap]: We don't wrap text without space before/after jsx expression, but we wrap multiple jsx expressions, don't we?

Comment (lydell) [reply tests/jsx-text-wrap/__snapshots__/jsfmt.spec.js.snap]: Ah, now I get it. Makes sense! :+1:

**Comment by lydell** [comment]
Thank you very much for doing this and sorry for the delay!
